csodiaq/identifier/outputWritingFunctions.py

```diff
@@ -0,0 +1,72 @@
+import pandas as pd
+
+def format_output_line(libMetadata, queMetadata, matchMetadata):
+    return [
+        queMetadata["scan"],
+        queMetadata["precursorMz"],
+        libMetadata["peptide"],
+        libMetadata["proteinName"],
+        libMetadata["precursorMz"],
+        libMetadata["precursorCharge"],
+        matchMetadata["cosineSimilarityScore"],
+        libMetadata["identifier"],
+        queMetadata["peaksCount"],
+        libMetadata["peaks"],
+        matchMetadata["shared"],
+        matchMetadata["ionCount"],
+        queMetadata["CV"],
+        queMetadata["windowWidth"],
+        matchMetadata["maccScore"],
+        #matchMetadata["excludeNum"],
+    ]
+
+def extract_metadata_from_match_dataframe_groupby(group):
+    return {
+        "shared": len(group.index),
+        "ionCount": sum(group["queryIntensity"]),
+        #TODO: see TODO in extract_metadata_from_match_and_score_dataframes test
+        #"extractNum": 0
+    }
+
+def extract_metadata_from_score_dataframe(df):
+    #TODO: there has to be a cleaner way to do this
```

Comment:
In a nutshell, I want to make a dictionary with the following shape:

key: (libraryIdx, queryIdx)
value: {
   'maccScore': X,
   'cosineScore':X,
}

Each row is unique, so used the pandas to_dict to convert the scoreDf (with columns 'libraryIdx','queryIdx','maccScore','cosineScore') to basically make 2 dictionaries corresponding to maccScore and cosineScore, respectively, before smooshing them together into one dictionary.

I may make a stack overflow comment on this - I wasn't finding anything that answered this specific question.